src/main/java/seedu/address/model/person/Status.java

```diff
@@ -29,24 +32,26 @@ public String getValue() {
      * Translate a string into a Status enum if the string matches any Status values. Trims string.
      *
      * @param string String to be translated.
-     * @return The translated Status if the string is valid, null object otherwise.
+     * @return The translated Status if the string is valid.
+     * @throws IllegalArgumentException if the string is invalid.
      */
-    public static Status translateStringToStatus(String string) {
+    public static Status translateStringToStatus(String string) throws IllegalArgumentException {
         String trimmedString = string.trim();
         Status resultStatus = null;
         for (Status r : Status.values()) {
             if (r.getValue().equalsIgnoreCase(trimmedString)) {
                 resultStatus = r;
             }
         }
-        return resultStatus;
+        if (Objects.isNull(resultStatus)) {
```

Comment:
Not to sure aboud using isNull as a null check, I think a reference check would suffice. Else clause is unnecessary here, could just keep the if clause to make a guard clause.